easybuild/easyblocks/g/gromacs.py

```diff
@@ -203,6 +205,65 @@ def configure_step(self):
+        cp2k_root = get_software_root('CP2K')
+        if self.cfg['cp2k'] and not cp2k_root:
+            msg = "CP2K support has been requested but CP2K is not listed as a dependency."
+            raise EasyBuildError(msg)
+        elif cp2k_root and self.cfg['cp2k'] is False:
+            self.log.info('CP2K was found, but compilation without CP2K has been requested.')
+            cp2k_root = None
+
+        if cp2k_root:
+            if LooseVersion(self.version) < LooseVersion('2022'):
+                msg = 'CP2K support is only available for GROMACS 2022 and newer.'
+                raise EasyBuildError(msg)
+            elif LooseVersion(get_software_version('CP2K')) < LooseVersion('8.1'):
+                msg = 'CP2K support in GROMACS requires CP2K version 8.1 or higher.'
+                raise EasyBuildError(msg)
+
+            if not self.cfg['mpi_only']:
+                msg = "GROMACS with CP2K support needs to be built with 'mpi_only = True'"
+                raise EasyBuildError(msg)
+
+            if not os.path.exists(os.path.join(cp2k_root, 'lib', 'libcp2k.a')):
+                msg = 'CP2K needs to be compiled with "library = True".'
+                raise EasyBuildError(msg)
+            if not os.path.exists(os.path.join(cp2k_root, 'lib', 'pkgconfig', 'libcp2k.pc')):
+                msg = "pkg-config is required as a build-dependency for CP2K"
+                raise EasyBuildError(msg)
+            if not get_software_root('pkg-config'):
+                msg = "pkg-config is required as a build-dependency for building GROMACS-CP2K"
+                raise EasyBuildError(msg)
+
+            self.log.info('CP2K support has been enabled.')
+            # Building with CP2K requires static build w/o gmxapi.
+            # https://manual.gromacs.org/documentation/2022/install-guide/index.html#building-with-cp2k-qm-mm-support
+            self.log.info("Building with CP2K QM/MM.")
+            self.cfg['build_shared_libs'] = False
+            self.libext = 'a'
+            cp2k_version = get_software_version('CP2K')
+            self.cfg.update('configopts', "-DGMX_INSTALL_NBLIB_API=OFF")
+            self.cfg.update('configopts', "-DGMXAPI=OFF")
+            self.cfg.update('configopts', "-DGMX_CP2K=ON")
+            # Ensure that the GROMACS log files report that CP2K was enabled and which version was used.
+            self.cfg.update('configopts', "-DGMX_VERSION_STRING_OF_FORK=CP2K-{:}".format(cp2k_version))
+            self.cfg.update('configopts', "-DCP2K_DIR=%s/lib64" % cp2k_root)
+            cp2k_linker_flags = [
+                # Need MPI linker flags b/c libcp2k.a is compiled with mpifort.
+                # These are for OpenMPI (mpifort --showme).
+                "-lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh",
+                "-L%s/lib/exts/dbcsr" % cp2k_root,
+                # get depenencies for libcp2k.a:
+                "$(pkg-config --libs-only-l libcp2k)"
+            ]
```

Comment:
This is not good. We must not depend on it being built with OpenMPI. To do it this way it MUST first check that OpenMPI is part of the toolchain.
And the pkgconfig for libcp2k should probably include the MPI libs it depends on which "should" then make all this redundant if the CMake for GROMACS does things correctly.

Reply:
Unfortunately CP2K's pkgconfig doesn't contain the MPI libs. That's why I had to hack them in manually.

```
$ pkg-config --libs-only-l libcp2k
-lcp2k -ldbcsr -lm -lpthread -lstdc++ -lxcf03 -lxc -lxsmm -lxsmmf -lopenblas -lgfortran -lfftw3_omp -lfftw3 -lpthread -lopenblas -lgfortran -lpthread -lscalapack -lopenblas -lgfortran
```
Should I rework it like this?

```python
cp2k_linker_flags = []
# Need MPI linker flags b/c libcp2k.a is compiled with mpifort.
# These are for OpenMPI (mpifort --showme).
if get_software_root('OpenMPI'):
    cp2k_linker_flags.append("-lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh")
elif get_software_root('IntelMPI'):
    # append Intel MPI libs for fortran here:
    cp2k_linker_flags.append("...")

cp2k_linker_flags += [
    "-L%s/lib/exts/dbcsr" % cp2k_root,
    # get dependencies for libcp2k.a:
    "$(pkg-config --libs-only-l libcp2k)"
]

```

Reply:
In 2bacbb3fc the easyblock is now checking for the cases of Open MPI and Intel MPI to add the fortran-MPI libraries.
Other MPI implementations could be added if needed.